File: components/events/TweetEditedEvent.ts
import { Environment, fetchQuery, graphql } from "relay-runtime"

import { createEventHook } from "components/EventsProvider"

export interface TweetEditedEvent {
  user_id: string
  item_id: string
}

const fetchTweetQuery = graphql`
  query TweetEditedEventQuery($id: ID!) {
    node(id: $id) {
      id
      ... on TweetGroup {
        tweets {
          body
          mediaURLs
        }
      }
    }
  }
`

export async function handleTweetEdited(
  environment: Environment,
  event: TweetEditedEvent
): Promise<void> {
  await fetchQuery(environment, fetchTweetQuery, { id: event.item_id })
}

export const useTweetEditedEvent = createEventHook(
  "TweetEdited",
  handleTweetEdited
)

File: ios/Courier/Courier/Events/TweetEditedEvent.swift
import Combine
import Relay
import CourierGenerated

struct TweetEditedEvent: Decodable {
    var userID: String
    var itemID: String

    enum CodingKeys: String, CodingKey {
        case userID = "user_id"
        case itemID = "item_id"
    }
}

private let query = graphql("""
query TweetEditedEventQuery($id: ID!) {
  tweetGroup(id: $id) {
    id
    tweets {
      body
      mediaURLs
    }
  }
}
""")

func handleTweetEdited(_ environment: Environment, _ event: TweetEditedEvent) -> AnyPublisher<Never, Error> {
    environment.fetchQuery(TweetEditedEventQuery(id: event.itemID))
        .ignoreOutput()
        .eraseToAnyPublisher()
}
